### DISCUSSION TITLE
How to measure the MessagesOutPerSec in Strimzi Kafka?

### DISCUSSION THREAD

**Discussion by KG-Anand** [strimzi/strimzi-kafka-operator#10772] [Q&A]
Currently, we have the below metric in JMX to get the Messages in rate at Topic level.
Mbean name: kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec,topic=([-.\w]+)

I cant able to locate the Messages out rate at Topic level.
Why its not exposed in strimzi kafka?
Do we have any backlog or KIP for the same?

How to we arrive the Messages out rate at Topic level that exactly matches with the Messages in rate.

Need your valuable answers from the community to get clarified, Thanks in advance.

**Comment by katheris** [comment]
Kafka doesn't provide a messages out rate metric, however you can use the BytesOutPerSec rate to track the flow of data to consumers
